pkg/odo/cli/component/create.go

```diff
@@ -36,282 +43,298 @@ var (
- 
+# Create new Node.js git component while specifying a branch, tag or commit ref
+%[1]s nodejs --git https://github.com/openshift/nodejs-ex.git --ref master
 
-  # Create a new Node.js component of version 6 from the 'openshift' namespace
-  odo create openshift/nodejs:6 --local /nodejs-ex
+# Create new Node.js git component while specifying a tag
+%[1]s nodejs --git https://github.com/openshift/nodejs-ex.git --ref v1.0.1
 
-  # Create new Wildfly component with binary named sample.war in './downloads' directory
-  odo create wildfly wildly --binary ./downloads/sample.war
 
-  # Create new Node.js component with the source in current directory and ports 8080-tcp,8100-tcp and 9100-udp exposed
-  odo create nodejs --port 8080,8100/tcp,9100/udp
+# Create a new Node.js component of version 6 from the 'openshift' namespace
+%[1]s openshift/nodejs:6 --local /nodejs-ex
 
-  # Create new Node.js component with the source in current directory and env variables key=value and key1=value1 exposed
-  odo create nodejs --env key=value,key1=value1
+# Create new Wildfly component with binary named sample.war in './downloads' directory
+%[1]s wildfly wildly --binary ./downloads/sample.war
 
-  # For more examples, visit: https://github.com/redhat-developer/odo/blob/master/docs/examples.md
-  odo create python --git https://github.com/openshift/django-ex.git
-	`,
-	Args: cobra.RangeArgs(1, 2),
-	Run: func(cmd *cobra.Command, args []string) {
+# Create new Node.js component with the source in current directory and ports 8080-tcp,8100-tcp and 9100-udp exposed
+%[1]s nodejs --port 8080,8100/tcp,9100/udp
 
-		stdout := color.Output
-		glog.V(4).Infof("Component create called with args: %#v, flags: binary=%s, git=%s, local=%s", strings.Join(args, " "), componentBinary, componentGit, componentLocal)
+# Create new Node.js component with the source in current directory and env variables key=value and key1=value1 exposed
+%[1]s nodejs --env key=value,key1=value1
 
-		context := genericclioptions.NewContextCreatingAppIfNeeded(cmd)
-		client := context.Client
-		projectName := context.Project
-		applicationName := context.Application
+# For more examples, visit: https://github.com/redhat-developer/odo/blob/master/docs/examples.md
+%[1]s python --git https://github.com/openshift/django-ex.git
+  `)
 
-		checkFlag := 0
-		componentPath := ""
-		var componentPathType occlient.CreateType
+// NewCreateOptions returns new instance of CreateOptions
+func NewCreateOptions() *CreateOptions {
+	return &CreateOptions{}
+}
 
-		if len(componentBinary) != 0 {
-			componentPath = componentBinary
-			path, err := util.GetAbsPath(componentPath)
-			odoutil.LogErrorAndExit(err, "Failed to resolve %s to absolute path", componentPath)
-			componentPath = path
-			componentPathType = occlient.BINARY
-			checkFlag++
-		}
-		if len(componentGit) != 0 {
-			componentPath = componentGit
-			componentPathType = occlient.GIT
-			checkFlag++
+func setCmpSourceAttrs(co *CreateOptions) (err error) {
```

Comment:
How about defining this function as a method of `CreateOptions`?

Reply:
Changed it now :)